lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -906,6 +906,8 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
 
     Value input = payloadArgs[0];
     Value mask = payloadArgs[1];
+    if (mask.getType().isa<mlir::FloatType>())
+      mask = b.create<arith::ConstantOp>(loc, b.getBoolAttr(false));
```

Comment:
I think I missed this the first time I reviewed your changes. Why is `mask` being set to `false` here?

Reply:
That seems to be expected behavior. I was casting to `Int1` at first, but further testing shows that it seems to treat all floats as false. I haven't found anything in the documentation about it.

Reply:
This seems to be a bug upstream. I would actually expect `float` mask to result in a runtime error, since this is the behavior that `aten.masked_select` has:

https://github.com/pytorch/pytorch/blob/5b58140d1a471b144baf66cc61a45a86746f0215/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L1720-L1721

Can you file a bug upstream for this?

Reply:
Okay, the bug is up [here](https://github.com/pytorch/pytorch/issues/85063). I'm going to leave this as-is for now in case it's expected behavior but I'll add an assert if it isn't.